src/proto/smb.rs

```diff
@@ -0,0 +1,1143 @@
+            }
+            SMB2SetupRequestState::Flags => {
+                self.flags = *byte;
+                self.d.next_state(SMB2SetupRequestState::SecurityMode);
+            }
+            SMB2SetupRequestState::SecurityMode => {
+                self.security_mode = *byte;
+                self.d.next_state(SMB2SetupRequestState::Capabilities);
+            }
+            SMB2SetupRequestState::Capabilities => {
+                self.capabilities =
+                    self.d
+                        .read_ule32(byte, self.capabilities, SMB2SetupRequestState::Channel);
+            }
+            SMB2SetupRequestState::Channel => {
+                self.channel = self.d.read_ule32(
+                    byte,
+                    self.channel,
+                    SMB2SetupRequestState::SecurityBufferOffset,
+                );
+            }
+            SMB2SetupRequestState::SecurityBufferOffset => {
+                self.security_buffer_offset = self.d.read_ule16(
+                    byte,
+                    self.security_buffer_offset,
+                    SMB2SetupRequestState::SecurityLen,
+                );
+            }
+            SMB2SetupRequestState::SecurityLen => {
+                self.security_len = self.d.read_ule16(
+                    byte,
+                    self.security_len,
+                    SMB2SetupRequestState::PreviousSessionId,
+                );
+            }
+            SMB2SetupRequestState::PreviousSessionId => {
+                self.previous_session_id =
+                    self.d
+                        .read_ule64(byte, self.previous_session_id, SMB2SetupRequestState::SecurityBlob);
+            }
+            SMB2SetupRequestState::SecurityBlob => {
+                // TODO ? Not super useful TBH, also this is ASN.1 :///
+                self.d.next_state(SMB2SetupRequestState::End);
+            },
+            SMB2SetupRequestState::End => {}
+        }
+    }
+
+    fn repl(&self) -> Option<Vec<u8>> {
+        None
+    }
+}
+
+#[derive(Debug, Clone)]
+enum SMB2Payload {
+    NegotiateRequest(SMB2NegotiateRequest),
+    SetupRequest(SMB2SetupRequest),
+}
+
+impl SMB2Payload {
+    fn repl(&self) -> Option<Vec<u8>> {
+        match self {
+            SMB2Payload::NegotiateRequest(x) => x.repl(),
+            SMB2Payload::SetupRequest(x) => x.repl(),
+        }
+    }
+    fn parse(&mut self, byte: &u8) {
+        match self {
+            SMB2Payload::NegotiateRequest(x) => x.parse(byte),
+            SMB2Payload::SetupRequest(x) => x.parse(byte),
+        }
+    }
+}
+
+//////////////
+// Handlers //
+//////////////
+
+pub fn repl_smb1<'a>(
+    data: &'a [u8],
+    _masscanned: &Masscanned,
+    _client_info: &ClientInfo,
+) -> Option<Vec<u8>> {
+    let mut nbtsession: NBTSession<SMB1Header> = NBTSession::new();
+    for byte in data {
+        nbtsession.parse(byte);
+    }
+    return nbtsession.repl();
```

Comment:
More rusty to use `nbtsession.repl()` (no `return`, no `;`) at the end of the function I think, do you agree?